plumber-android/src/main/java/leakcanary/AndroidLeakFixes.kt

```diff
@@ -198,12 +205,223 @@ enum class AndroidLeakFixes {
     }
   },
 
+  /**
+   * ActivityChooserModel holds a static reference to the last set ActivityChooserModelPolicy
+   * which can be an activity context.
+   * Tracked here: https://code.google.com/p/android/issues/detail?id=172659
+   */
+  ACTIVITY_CHOOSER {
+    override fun apply(application: Application) {
+      if (SDK_INT >= 28) {
+        return
+      }
+
+      backgroundExecutor.execute {
+        val activityChooserModelClass: Class<*>
+        val activityChooserGetterMethod: Method
+        val activityChooserSetterMethod: Method
+        try {
+          activityChooserModelClass = Class.forName("android.widget.ActivityChooserModel")
+          activityChooserGetterMethod = activityChooserModelClass
+            .getDeclaredMethod("get", Context::class.java, String::class.java)
+          activityChooserGetterMethod.isAccessible = true
+          val onChooseActivityListenerClass = activityChooserModelClass
+            .declaredClasses
+            .first { clazz -> clazz.simpleName == "OnChooseActivityListener" }
+          activityChooserSetterMethod = activityChooserModelClass
+            .getDeclaredMethod("setOnChooseActivityListener", onChooseActivityListenerClass)
+          activityChooserSetterMethod.isAccessible = true
+        } catch (ignored: Exception) {
+          return@execute
+        }
+
+        application.onActivityDestroyed {
+          try {
+            val activityChooser = activityChooserGetterMethod
+              .invoke(null, application, ShareActionProvider.DEFAULT_SHARE_HISTORY_FILE_NAME)
+            activityChooserSetterMethod.invoke(activityChooser, null)
+          } catch (ignored: Exception) {
+            // Silent
+          }
+        }
+      }
+    }
+  },
+
+  /**
+   * ConnectivityManager has a sInstance field that is set when the first ConnectivityManager instance is created.
+   * ConnectivityManager has a mContext field.
+   * When calling activity.getSystemService(Context.CONNECTIVITY_SERVICE) , the first ConnectivityManager instance
+   * is created with the activity context and stored in sInstance.
+   * That activity context then leaks forever.
+   *
+   * Tracked here: https://code.google.com/p/android/issues/detail?id=198852
+   * Introduced here: https://github.com/android/platform_frameworks_base/commit/e0bef71662d81caaaa0d7214fb0bef5d39996a69
+   */
+  CONNECTIVITY_MANAGER {
+    override fun apply(application: Application) {
+      if (SDK_INT > 23) {
+        return
+      }
+
+      try {
+        application.getSystemService(Context.CONNECTIVITY_SERVICE)
+      } catch (ignored: Exception) {
+        // Silent
```

Comment:
did this happen, or is it to be safe?

Reply:
As for the previous comment, it's based on `CONNECTIVITY_MANAGER__SINSTANCE` in `AndroidReferenceMatchers`, but I didn't see it happen in the wild, so it's pretty much to be safe